akka-docs/src/test/scala/typed/tutorial_4/Device.scala

```diff
@@ -0,0 +1,67 @@
+/*
+ * Copyright (C) 2009-2018 Lightbend Inc. <https://www.lightbend.com>
+ */
+
+package typed.tutorial_4
+
+//#device-with-passivate
+import akka.actor.typed.ActorRef
+import akka.actor.typed.Behavior
+import akka.actor.typed.PostStop
+import akka.actor.typed.Signal
+import akka.actor.typed.scaladsl.AbstractBehavior
+import akka.actor.typed.scaladsl.ActorContext
+import akka.actor.typed.scaladsl.Behaviors
+
+object Device {
+  def apply(groupId: String, deviceId: String): Behavior[DeviceMessage] =
+    Behaviors.setup(context ⇒ new Device(context, groupId, deviceId))
+
+  sealed trait DeviceMessage
+
+  final case class ReadTemperature(requestId: Long, replyTo: ActorRef[RespondTemperature])
+    extends DeviceMessage
+  final case class RespondTemperature(requestId: Long, value: Option[Double])
+
+  final case class RecordTemperature(requestId: Long, value: Double, replyTo: ActorRef[TemperatureRecorded])
+    extends DeviceMessage
+  final case class TemperatureRecorded(requestId: Long)
+
+  //#passivate-msg
+  case object Passivate extends DeviceMessage
+  //#passivate-msg
+}
+
+class Device(context: ActorContext[Device.DeviceMessage], groupId: String, deviceId: String)
+  extends AbstractBehavior[Device.DeviceMessage] {
+  import Device._
+
+  var lastTemperatureReading: Option[Double] = None
```

Comment:
I'm consistently only using the OO style in this guide, because I think most developers will find that more familiar than the functional style. Mentioning it in Next steps, though.